muffnn/mlp/base.py

```diff
@@ -301,6 +301,26 @@ def _set_up_graph(self):
         if self._transform_layer_index == -1:
             self._transform_layer = t
 
+        # Prediction gradients (e.g., for analyzing the importance of features)
+        # We use the top layer before the output activation function
+        # (e.g., softmax, sigmoid) following
+        # https://arxiv.org/pdf/1312.6034.pdf
+        if self.is_sparse_:
+            self._prediction_gradient = None
+        else:
+            output_shape = self.output_layer_.get_shape()
+            if len(output_shape) == 1:
+                self._prediction_gradient = tf.gradients(
+                    t, self._input_values)[0]
+            else:
```

Comment:
Bit of a stretch, but should this check for `len(output_shape) == 2`? The only case I can think of where it might matter is if you had a timeseries multiclass problem though so maybe not necessary.

Reply:
I don't think it's possible to have dimension > 2, but an extra check and valueerror wouldn't hurt. will add